exec-c-api/src/capi_instance_cache.rs

```diff
@@ -0,0 +1,84 @@
+use std::slice;
+
+use elrond_exec_service::CompilationOptions;
+
+use crate::{
+    capi_executor::{vm_exec_executor_t, CapiExecutor},
+    capi_instance::{vm_exec_compilation_options_t, vm_exec_instance_t, CapiInstance},
+    service_singleton::with_service,
+    vm_exec_result_t,
+};
+
+/// Caches an instance.
+///
+/// # Safety
+///
+/// C API function, works with raw object pointers.
+#[allow(clippy::cast_ptr_alignment)]
+#[no_mangle]
+pub unsafe extern "C" fn vm_exec_instance_cache(
+    instance_ptr: *const vm_exec_instance_t,
+    cache_bytes_ptr: *mut *const u8,
+    cache_bytes_len: *mut u32,
+) -> vm_exec_result_t {
+    let capi_instance = cast_input_const_ptr!(instance_ptr, CapiInstance, "instance ptr is null");
+
+    let result = capi_instance.content.cache();
+    match result {
+        Ok(bytes) => {
+            *cache_bytes_ptr = bytes.as_ptr();
+            *cache_bytes_len = bytes.len() as u32;
+            std::mem::forget(bytes);
```

Comment:
In order to deal with this memory leak later, it might be preferable to place the data in the Instance object, so that it gets cleaned with the instance.

Reply:
will address in next pr